eth_abi/decoding.py

```diff
@@ -24,28 +24,11 @@
 
 
 class BaseDecoder(BaseCoder):
-    @classmethod
-    def as_decoder(cls, name=None, **kwargs):
-        for key in kwargs:
-            if not hasattr(cls, key):
-                raise AttributeError(
-                    "Property `{key}` not found on {cls_name} class. "
-                    "`{cls_name}.as_decoder` only accepts keyword arguments which are "
-                    "present on the {cls_name} class.".format(
-                        key=key,
-                        cls_name=cls.__name__,
-                    )
-                )
-        if name is None:
-            name = cls.__name__
-        sub_cls = type(name, (cls,), kwargs)
-        sub_cls.validate()
-        instance = sub_cls()
-        return instance
-
-    @classmethod
-    def validate(cls):
-        pass
+    def decode(self, stream):  # pragma: no cover
```

Comment:
Now that we're instantiating these, maybe we should make use of `abc.ABCMeta` and `abc.abstractmethod`?

Reply:
Sure.  Made that change.